src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -1515,7 +1515,9 @@ void Transformations::MainSnippets() {
         [[maybe_unused]] const auto& inferencePrecision = config.inferencePrecision;
         // Note: BrgemmTPP doesn't support transposed KN natively
         // so we should extract transposes for the corresponding matmul nodes
-#if defined(SNIPPETS_LIBXSMM_TPP)
+#if defined(OPENVINO_ARCH_ARM64)
+        return false;
+#elif defined(SNIPPETS_LIBXSMM_TPP)
```

Comment:
Since we have opportunity to enable libxsmm on aarch64 and execute brgemm via this backend, could you please move arch-related blocks after `defined(SNIPPETS_LIBXSMM_TPP)`? And for non arm64 and x64 let's thow exception to not forget to update this callback in the future
```
#if defined(SNIPPETS_LIBXSMM_TPP)
.....
#endif
#if defined(OPENVINO_ARCH_ARM64)
   return false;
#eliif defined(OPENVINO_ARCH_X86_64)
   return true;
#else
   throw_exception()
#endif
```